File: sdk-python/src/c6_guardrails/types/cq_scan.py
from dataclasses import dataclass, field
from typing import Literal


@dataclass
class CqScanRequest:
    input: str
    response: str


@dataclass
class CqScanResult:
    object: str = ""  # "firewall.cq_scan"
    request_id: str = ""
    groundedness: list[float] = field(default_factory=list)
    relevance: list[float] = field(default_factory=list)
    hallucination: list[float] = field(default_factory=list)
    verdict: Literal["allow", "flag", "block"] = "allow"
    action: str = ""
    reason: str = ""
    duration_ms: float = 0.0

File: sdk-node/src/types/cq_scan.ts
export interface CqScanRequest {
  input: string;
  response: string;
}

export interface CqScanResult {
  object: 'firewall.cq_scan';
  request_id: string;
  groundedness: number[];
  relevance: number[];
  hallucination: number[];
  verdict: 'allow' | 'flag' | 'block';
  action: string;
  reason: string;
  duration_ms: number;
}
